File: saksi_frontend/src/application/usecases/start_session.ts
import type { SessionRepository } from '../../domain/repositories/session_repository'
import type { Session } from '../../domain/entities/session'

export class StartSessionUseCase {
  private readonly repo: SessionRepository

  constructor(repo: SessionRepository) {
    this.repo = repo
  }

  async execute(officerId: string, productId: string): Promise<Session> {
    if (!officerId.trim()) throw new Error('ID petugas wajib diisi')
    return this.repo.start(officerId, productId)
  }
}

File: saksi_mobile/lib/features/session/domain/usecases/start_session.dart
import '../entities/session.dart';
import '../repositories/session_repository.dart';

class StartSessionUseCase {
  const StartSessionUseCase(this._repo);
  final SessionRepository _repo;

  Future<Session> call(String officerId, String productId) {
    if (officerId.trim().isEmpty) {
      throw ArgumentError('ID petugas wajib diisi');
    }
    return _repo.start(officerId, productId);
  }
}
